lib/application/widgets/theme/rich_text_field.dart

```diff
@@ -0,0 +1,343 @@
+      final hasAttribute = selectedAttributes.value.any((currentAttribute) => currentAttribute == attribute);
+
+      if (hasAttribute) {
+        controller.formatSelection(quill.Attribute.clone(attribute, null));
+      } else {
+        controller.formatSelection(attribute);
+      }
+    }
+
+    bool isSelected(quill.Attribute attribute) => selectedAttributes.value.contains(attribute);
+
+    final attributesIcons = _toolBarAsset.keys
+        .map((attribute) => AssetIconButton(
+              _toolBarAsset[attribute]!,
+              iconColor: isSelected(attribute) ? theme.neutralSwatch.shade800 : null,
+              iconBackgroundColor: isSelected(attribute) ? theme.neutralSwatch.shade500 : null,
+              isSplashEffectEnabled: false,
+              onPressed: () => onToggle(attribute),
+            ))
+        .toList();
+
+    return ThemedBottomContainer(
+      child: Container(
+        color: theme.neutralSwatch.shade800,
+        child: SafeArea(
+          child: Row(mainAxisAlignment: MainAxisAlignment.center, children: attributesIcons),
+        ),
+      ),
+    );
+  }
+}
+
+const _useQuillController = _QuillControllerHookCreator();
+
+class _QuillControllerHookCreator {
+  const _QuillControllerHookCreator();
+
+  /// Creates a [quill.QuillController] that will be disposed automatically.
+  ///
+  /// The [textController] parameter can be used to set the initial value of the controller text ands its selection.
+  quill.QuillController call({TextEditingController? textController, List<Object?>? keys}) =>
+      use(_QuillControllerHook(textController: textController, keys: keys));
+}
+
+class _QuillControllerHook extends Hook<quill.QuillController> {
+  const _QuillControllerHook({this.textController, List<Object?>? keys = const []}) : super(keys: keys);
+
+  final TextEditingController? textController;
+
+  @override
+  _QuillControllerHookState createState() => _QuillControllerHookState();
+}
+
+class _QuillControllerHookState extends HookState<quill.QuillController, _QuillControllerHook> {
+  late final quill.QuillController _controller;
+
+  @override
+  void initHook() {
+    final text = hook.textController?.text;
+    final hasText = text != null && text.isNotEmpty;
+
+    final selection = hook.textController?.selection;
+    final hasSelection = selection != null && selection.isValid;
+
+    final document = hasText ? quill.Document.fromJson(json.decode(text!) as List<dynamic>) : quill.Document();
+    final selectionOffset = document.toPlainText().isNotEmpty ? document.toPlainText().length - 1 : 0;
+
+    _controller = quill.QuillController(
+      document: document,
+      selection: hasSelection ? selection! : TextSelection.collapsed(offset: selectionOffset),
+    );
+  }
+
+  @override
+  quill.QuillController build(BuildContext context) => _controller;
+
+  @override
+  void dispose() => _controller.dispose();
+
+  @override
+  String get debugLabel => 'useQuillController';
+}
```

Comment:
Created a custom hook to control `quill.QuillController` lifecycle in the StatelessWidget. What do you think?